sources/HUBViewModelDiff.m

```diff
@@ -57,7 +64,9 @@ + (instancetype)diffFromViewModel:(id<HUBViewModel>)fromViewModel
                         algorithm:(HUBDiffAlgorithm)algorithm
 {
     NSParameterAssert(algorithm);
-    return algorithm(fromViewModel, toViewModel);
+    HUBViewModelDiff *diff =  algorithm(fromViewModel, toViewModel);
+    [diff calculateHeaderChangesFromViewModel:fromViewModel toViewModel:toViewModel];
+    return diff;
```

Comment:
What about when we include overlays in the diffing process (which we really should do). Wouldn't we want to reuse the diffing functions we already have for body components - i.e. remove the body specifics, make them more generic, then reuse them for the overlay components.

If we do that, considering the header components in those functions doesn't make sense either.